helm/generic-node/templates/generic-node.yaml

```diff
@@ -0,0 +1,283 @@
+        - name: DAEMON_METRICS_PORT
+          value: {{.Values.ports.metrics | quote}}
+        - name: MINA_CLIENT_TRUSTLIST
+          value: "10.0.0.0/8"
+        - name: CODA_CLIENT_TRUSTLIST
+          value: "10.0.0.0/8"
+        - name: CONNECT_PRIVATE_IPS
+          value: "true"
+        - name: GCLOUD_KEYFILE
+          value: "/gcloud/keyfile.json"
+        - name: NETWORK_NAME
+          value: {{ .Values.testnetName}}
+        - name: GCLOUD_BLOCK_UPLOAD_BUCKET
+          value: "features_network_block_data"
+        - name: DAEMON_EXTERNAL_PORT
+          value: {{.Values.ports.p2p | quote}}
+        - name: CODA_SNARK_KEY
+          value: {{ $.Values.publicKey | quote }}
+        - name: CODA_SNARK_FEE
+          value: {{ $.Values.snarkFee | quote }}
+        - name: WORK_SELECTION
+          value: {{$.Values.workSelectionAlgorithm | quote }}
+        - name: "RESTART_EVERY_MINS"
+          value: {{ .Values.restartEveryMins | quote }}
+        - name: "RESTART_NODES"
+          value: {{ .Values.restartNodes | quote }}
+        - name: "SEED_PEERS"
+          value: {{ $.Values.daemon.seedPeers | quote }}
+        - name: "SEED_PEERS_URL"
+          value: {{ $.Values.seedPeersURL | quote }}
+        {{- if .Values.libp2p.staticKeypair.enabled}}
+        - name: CODA_LIBP2P_PASS
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.libp2p.staticKeypair.secretName | quote}}
+              key: password
+        - name: MINA_LIBP2P_PASS
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.libp2p.staticKeypair.secretName | quote}}
+              key: password
+        {{- end }}
+        {{- if .Values.features.privateKeySecret.enabled}}
+        - name: CODA_PRIVKEY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.features.privateKeySecret.secretName | quote}}
+              key: password
+        - name: MINA_PRIVKEY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.features.privateKeySecret.secretName | quote}}
+              key: password
+        {{- end }}
+        ports:
+        - name: client-port
+          protocol: TCP
+          containerPort: {{.Values.ports.client}}
+        - name: graphql-port
+          protocol: TCP
+          containerPort: {{.Values.ports.graphql}}
+        - name: metrics-port
+          protocol: TCP
+          containerPort: {{.Values.ports.metrics}}
+        - name: external-port
+          protocol: TCP
+          containerPort: {{.Values.ports.p2p}}
+          hostPort: {{.Values.ports.external}}
+      volumes:
+      - name: destination-libp2p
+        emptyDir: {}
+      - name: config-dir
+        emptyDir: {}
+      - name: actual-libp2p
+        emptyDir: {}
+      - name: wallet-keys
+        emptyDir: {}
+      {{- if .Values.features.runtimeConfig.enabled}}
+      - name: daemon-config
+        configMap:
+          name: seed-daemon-config
```

Comment:
this secret name should be configurable for example, `.Values.features.runtimeConfig.secretName`